examples/app/headless_renderer.rs

```diff
@@ -0,0 +1,506 @@
+//! 5) Exit if `single_image` setting is set
+
+use bevy::{
+    app::ScheduleRunnerPlugin, core_pipeline::tonemapping::Tonemapping, prelude::*,
+    render::renderer::RenderDevice,
+};
+
+use crossbeam_channel::{Receiver, Sender};
+
+// To communicate between the main world and the render world we need a channel.
+// Since the main world and render world run in parallel, there will always be a frame of latency
+// between the data sent from the render world and the data received in the main world
+//
+// frame n => render world sends data through the channel at the end of the frame
+// frame n + 1 => main world receives the data
+//
+// Receiver and Sender are kept in resources because there is single camera and single target
+// That's why there is single images role, if you want to differentiate images
+// from different cameras, you should keep Receiver in ImageCopier and Sender in ImageToSave
+// or send some id with data
+
+/// This will receive asynchronously any data sent from the render world
+#[derive(Resource, Deref)]
+struct MainWorldReceiver(Receiver<Vec<u8>>);
+
+/// This will send asynchronously any data to the main world
+#[derive(Resource, Deref)]
+struct RenderWorldSender(Sender<Vec<u8>>);
+
+// Parameters of resulting image
+struct AppConfig {
+    width: u32,
+    height: u32,
+    single_image: bool,
+}
+
+fn main() {
+    let config = AppConfig {
+        width: 1920,
+        height: 1080,
+        single_image: true,
+    };
+
+    // setup frame capture
+    App::new()
+        .insert_resource(frame_capture::scene::SceneController::new(
+            config.width,
+            config.height,
+            config.single_image,
+        ))
+        .insert_resource(ClearColor(Color::srgb_u8(0, 0, 0)))
+        .add_plugins(
+            DefaultPlugins
+                .set(ImagePlugin::default_nearest())
+                .set(WindowPlugin {
+                    primary_window: None,
+                    exit_condition: bevy::window::ExitCondition::DontExit,
+                    close_when_requested: false,
+                }),
+        )
+        .add_plugins(frame_capture::image_copy::ImageCopyPlugin)
+        // headless frame capture
+        .add_plugins(frame_capture::scene::CaptureFramePlugin)
+        .add_plugins(ScheduleRunnerPlugin::run_loop(
+            std::time::Duration::from_secs_f64(1.0 / 60.0),
+        ))
+        .init_resource::<frame_capture::scene::SceneController>()
+        .add_systems(Startup, setup)
+        .run();
+}
+
+fn setup(
+    mut commands: Commands,
+    mut images: ResMut<Assets<Image>>,
+    mut scene_controller: ResMut<frame_capture::scene::SceneController>,
+    render_device: Res<RenderDevice>,
+) {
+    let render_target = frame_capture::scene::setup_render_target(
+        &mut commands,
+        &mut images,
+        &render_device,
+        &mut scene_controller,
+        15,
+        String::from("main_scene"),
```

Comment:
```suggestion
        "main_scene".into(),
```

Type elision can conclude that this is a `String` type :)